NETDefault/FunctionAppNETCore31/Models/ExampleSettings.cs

```diff
@@ -7,6 +7,6 @@ public class ExampleSettings
     {
         [Required]
         [MinLength(10)]
-        public string Example { get; set; }
+        public string Example { get; set; } = default!;
```

Comment:
I don't really agree to using the default keyword for non generics. 
Also, this is null forgiving a null literal, which is probably not what you want to be doing.

Reply:
Well, this is for now the only way to get this to work without having a `string?` or a warning. Changing this to `string?` is not really useful since the `[Required]` makes sure that the settings class when created, filled with config and then validated will be not null. If you new `ExampleSettings` out of the blue you can still get a `null`, which will probably be fixed in C# 10 where we can specify `public required string Example`, but I'm not sure if that can be used with deserializers. 

As I said further down, this is a bit of a pinky promise, but the developer communicates that the value will not be `null` when the object is used as intended.

Reply:
Oka, fair enough on the second point, but I would still suggest using null! over default! wherever possible.

Reply:
To me it looks roughly as silly as this:
```            for (int i = default; i < 100; i++)
            { }```

Reply:
Nullable reference types are not waterproof yet, but can be a big help already. So far I've experienced that you need to be very strict in assuring all types contain a value, for otherwise you'll be flooded with warnings.

In CitizenM.Subscriptions we used the `default!` operator for RequestDto objects, which went through the model validator, and so we could ensure these classes could not be null inside our business logic. 
`?`, `string.Empty` and others we used in the remaining cases, but always assuring that they could not be null. 

I think consistency is key here.